csrc/ir/nodes.cpp

```diff
@@ -5364,7 +5364,9 @@ std::vector<PolymorphicValue> SdpaBwdOp::evaluate(
   // Backward tensor inputs: grad_input, query, key, value, output,
   // logsumexp, max_q/k Temporary handling of DID parallelization. See
   // https://github.com/NVIDIA/Fuser/issues/2563
-  bool first_dim_is_did = this->key()->as<TensorView>()->axis(0)->isDeviceDim();
+  auto query_domain =
+      TensorDomain::noReductions(this->query()->getLogicalDomain());
```

Comment:
The loop domain starts with an `r`? I'm not saying this is impossible but I'm curious what this->query() looks like and how this is going to be affected #4381